### PR TITLE
Various Travis CI and related issues (closes #157, #250)

### LINKED ISSUE #157: Enhance Travis CI job with more validity checks
Lets consider adding pyflake to our travis ci job description. That would help us even more to have sane code in our repository.

### PR DESCRIPTION
**Pull request by TLemur**
Please merge, it's ready now.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,10 @@
 language: python
 python:
-  - "2.6"
   - "2.7"
 install:
-  - pip install pep8
+  - pip install pylama
```
Comment (whimboo) [inline .travis.yml]: Nice one! I haven't known about this tool. It looks great that it wraps a couple of tools into a single one.

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,10 @@
 language: python
 python:
-  - "2.6"
   - "2.7"
```
Comment (whimboo) [inline .travis.yml]: So this fixes issue #250.

File: mozdownload/__init__.py [hunk 3]
```diff
@@ -2,4 +2,4 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from scraper import *
+from scraper import * #noqa
```
Comment (whimboo) [inline mozdownload/__init__.py]: Please get this comment removed.

Comment (TLemur) [inline mozdownload/__init__.py]: If I remove the comment, it won't pass one of pylama's check

File: .travis.yml (hunk 2 above)
Comment (TLemur) [inline .travis.yml]: Both #250 and #157

File: mozdownload/__init__.py (hunk 3 above)
Comment (whimboo) [inline mozdownload/__init__.py]: I assume it's because of the `import *`? Maybe we should simply state every supported scraper class.

Comment (TLemur) [inline mozdownload/__init__.py]: I tried this as well. But then pylama complains they are not used.

Comment (whimboo) [inline mozdownload/__init__.py]: Given that wild-card imports are considered bad, you can import each scraper class individually and add them to a `__all__` array.

**Comment by TLemur** [comment]
Given that it is passing Travis CI, I don't doubt it can be merged now.

File: mozdownload/__init__.py [hunk 4]
```diff
@@ -2,4 +2,15 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from scraper import *
+from scraper import PLATFORM_FRAGMENTS, NotSupportedError, NotFoundError, \
+    NotImplementedError, TimeoutError, Scraper, DailyScraper, \
+    DirectScraper, ReleaseScraper, ReleaseCandidateScraper, \
+    TinderboxScraper, TryScraper
```
Comment (whimboo) [inline mozdownload/__init__.py]: I would suggest that we limit the number of exported symbols to what external code really needs. That means please remove `PLATFORM_FRAGMENTS` which is used only internally. For the others I would suggest to use one per line, and all surrounded by brackets, so you can avoid the backslashes. See the following link as an example: https://github.com/mozilla/firefox-ui-tests/blob/master/firefox_ui_harness/runtests.py#L13

File: mozdownload/__init__.py [hunk 5]
```diff
@@ -2,4 +2,15 @@
...
+from parser import DirectoryParser
+from timezones import PacificTimezone
```
Comment (whimboo) [inline mozdownload/__init__.py]: As a normal user you don't need those. If necessary they need to be imported when needed.

File: mozdownload/__init__.py (hunk 4 above)
Comment (TLemur) [inline mozdownload/__init__.py]: I agree that as a normal user you do not need PLATFORM_FRAGMENTS, but tests fail is it's removed.

File: mozdownload/__init__.py (hunk 5 above)
Comment (TLemur) [inline mozdownload/__init__.py]: Again, tests fail.

File: mozdownload/__init__.py (hunk 4 above)
Comment (whimboo) [inline mozdownload/__init__.py]: Tests should import this constant via `from mozdownload.scraper import PLATFORM_FRAGMENTS`. That will still be allowed.

File: mozdownload/__init__.py (hunk 5 above)
Comment (whimboo) [inline mozdownload/__init__.py]: Maybe you are seeing #293? It's a perma fail for me locally.

**Comment by whimboo** [comment]
I went ahead and fixed the remaining issues in PR #296. So I'm going to close this out.

**Pull request closed without merging**